Document/0x05d-Testing-Data-Storage.md

```diff
@@ -1069,166 +1069,244 @@ setContentView(R.layout.activity_main);
+Static analysis will help you identify potential problems, but it can not provide you with statistics on how long certain data is exposed in memory, nor it can help you identify problems in closed-source dependencies. This is where dynamic analysis comes into play.
+
+There are basically two ways to analyze the memory of a process: live analysis over a debugger or by analyzing one or more memory dumps. As the first approach is more of a general debugging issue, we concentrate on the second one here.
+
+For rudimentary analysis you can use the built-in tools of Android Studio. They are included under the _Android Monitor_ tab. To make a memory dump, select the device and app you want to analyze and click on _Dump Java Heap_. This will create a _.hprof_ file in the _captures_ directory relative to the project path of the app.
 
 ![Create Heap Dump](Images/Chapters/0x05d/Dump_Java_Heap.png)
 
-In the new tab that shows the _.hprof_ file, the "Package Tree View" should be selected. Afterwards the package name of the app can be used to navigate to the instances of classes that were saved in the memory dump.
+To navigate trough class instances saved in the memory dump, select the Package Tree View in the tab showing the _.hprof_ file.
 
 ![Create Heap Dump](Images/Chapters/0x05d/Package_Tree_View.png)
 
-The _.hprof_ file will be stored in the directory "captures", relative to the project path open within Android Studio. For deeper analysis of the memory dump the tool Eclipse Memory Analyzer (MAT) should be used.
+For more advanced analysis over the memory dump, Eclipse Memory Analyzer (MAT) can be used. It is available either as an Eclipse plugin or as a standalone application.
 
-Before the _.hprof_ file can be opened in MAT it needs to be converted. The tool _hprof-conf_ can be found in the Android SDK in the directory platform-tools.
+In order to be able to analyze the dump in MAT you need to use the _hprof-conv_ platform tool, provided with the Android SDK. 
 
 ```bash
-./hprof-conv file.hprof file-converted.hprof
+./hprof-conv memory.hprof memory-mat.hprof
 ```
 
-By using MAT, more functions are available, like usage of the Object Query Language (OQL). OQL is an SQL-like language that can be used to make queries in the memory dump. Analysis should be done on the dominator tree as only this contains the variables/memory of static classes.
+MAT provides several different tools you can use to analyze the memory dump. For example, you can use the _Histogram_ to get an idea on how many objects have been captured from a certain type, or the _Thread Overview_ to see process' threads and their stack frames. Check the _Dominator Tree_ to learn about keep-alive dependencies between objects. (**TO BE CONFIRMED OR REMOVED:** The _Dominator Tree_ is also the only tool containing the information for static classes). You can use regular expressions to filter out the results in all of these tools.
```

Comment:
I think someone told me this during a conference a while ago, that in the dominator tree you can find information about the static classes and methods. I would say delete this sentence, as during analysis you are looking for certain patterns, regardless if they are from a static class or not.